### PR TITLE
chore(cli): env init add env to stackset before env stack creation

### PR DESCRIPTION
**Pull request by iamhopaul123**
Currently we create the env stack before adding the env to the app stack set, which means the ECR repo will be created after the env stack creation. This becomes a problem when we want to upload the env stack custom resources script to the s3 bucket created by the app stack set.

With this PR, we swap the order so that the env will be added to the stackset before creating the env stack, making it possible to upload the custom resource scripts to the app s3 bucket.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/cli/env_init.go [hunk 1]
```diff
@@ -141,7 +141,7 @@ type initEnvOpts struct {
 	prompt       prompter
 	selVPC       ec2Selector
 	selCreds     credsSelector
-	selApp 	     appSelector
+	selApp       appSelector
```
Comment (Lou1415926) [inline internal/pkg/cli/env_init.go:144]: lol why is this showing up as diff...

Edit: sorry I should have finished all reviews before sending this comment out

Comment (iamhopaul123) [reply internal/pkg/cli/env_init.go:144]: Haha no worries it's totally fine.

File: internal/pkg/cli/env_init.go [hunk 2]
```diff
@@ -229,34 +229,45 @@ func (o *initEnvOpts) Execute() error {
...
+	if err := o.addToStackset(&deploycfn.AddEnvToAppOpts{
+		App:          app,
+		EnvName:      o.name,
+		EnvRegion:    aws.StringValue(o.sess.Config.Region),
+		EnvAccountID: envCaller.Account,
+	}); err != nil {
+		return err
+	}
+
+	// 2. Attempt to create the service linked role if it doesn't exist.
```
Comment (efekarakus) [inline internal/pkg/cli/env_init.go]: can we keep this as the very first thing that we do? The service linked role is the minimum thing that they need to be able to at least use ECS

File: internal/pkg/cli/env_init.go [hunk 3]
```diff
@@ -515,34 +526,29 @@ func (o *initEnvOpts) deployEnv(app *config.Application) error {
...
 	return nil
 }
 
-func (o *initEnvOpts) delegateDNSFromApp(app *config.Application) error {
-	envAccount, err := o.envIdentity.Get()
-	if err != nil {
-		return fmt.Errorf("getting environment account ID for DNS Delegation: %w", err)
-	}
-
+func (o *initEnvOpts) delegateDNSFromApp(app *config.Application, envAccount string) error {
```
Comment (efekarakus) [inline internal/pkg/cli/env_init.go]:
```suggestion
func (o *initEnvOpts) delegateDNSFromApp(app *config.Application, accountID string) error {
```

File: internal/pkg/deploy/cloudformation/app.go [hunk 4]
```diff
@@ -258,14 +258,22 @@ func (cf CloudFormation) removeWorkloadFromApp(app *config.Application, wlName s
...
 }
 
+// AddEnvToAppOpts contains the parameters to call AddEnvToApp.
+type AddEnvToAppOpts struct {
+	App          *config.Application
+	EnvName      string
+	EnvAccountID string
+	EnvRegion    string
+}
+
```
Comment (Lou1415926) [inline internal/pkg/deploy/cloudformation/app.go:268]: nice!!

**Review by Lou1415926 — COMMENTED** [review]
The code looks good to me!

Was there any reason why we did it the way it was before (add to app stackset after creating the stack)?

**Review by huanjani — APPROVED** [review]
LGTM!

**Review by efekarakus — APPROVED** [review]

**Pull request merged**